qiskit/visualization/timeline/core.py

```diff
@@ -159,6 +159,7 @@ def load_program(self, program: circuit.QuantumCircuit):
                 "This circuit should be transpiled with scheduler though it consists of "
                 "instructions with explicit durations.",
                 DeprecationWarning,
+                stacklevel=3,
```

Comment:
Just to confirm, the use of `stacklevel=3` is so that the warning points to `timeline_drawer`, correct? Asking because most of our deprecation warnings are `stacklevel=2`.

Reply:
Yeah, in this case it's a higher stack level because the function that triggers the warning is a "helper" function to the public API, so we need to unwind one extra stack frame before blaming it.  The idea is that `warnings.warn(stacklevel=n)` blames the code `n` calls above `warnings.warn` for being wrong.  So `stacklevel=1` (the default) blames the caller of `warn`, and `stacklevel=2` blames the caller of the caller of `warn`.  In public-library code, that's usually the API entry point, because it's usually entry points' jobs to normalise inputs and issue deprecation warnings.  When we use a helper function to unify the warning logic, the stack level often rises to 3, because the entry point calls the helper, which calls `warn`, and it was still the user's fault.

In this case, `load_program` is _kind of_ public API itself, but in practice the user entry point is almost invariable `timeline.draw`.  The worst that happens if the stack level is too high, though, is that some other code a level higher than the user's function (which is almost certainly still user code) gets blamed instead, so it should still show.